libazureinit/src/provision/password.rs

```diff
@@ -29,6 +31,14 @@ impl Provisioner {
 
 #[instrument(skip_all)]
 fn passwd(user: &User) -> Result<(), Error> {
+    // Update the sshd configuration to allow password authentication.
+    let ret = update_sshd_config();
+    if ret.is_err() {
+        return Err(Error::SubprocessFailed {
+            command: "update_sshd_config".to_string(),
+            status: Default::default(),
+        });
+    }
```

Comment:
As far as I understand, `SubprocessFailed` is supposed to be used for dealing with errors when running a separate command-line tool, like `hostname`, `ls` etc.
In this case, since a command-line tool `update_sshd_config` does not exist, I would rather recommend adding a new error type, e.g. `Error::UpdateSshdConfig`.